### DISCUSSION TITLE
Configure MongoDB URL when using Github + Vault

### DISCUSSION THREAD

**Discussion by swageblock** [mia-platform/community#506] [Q&A] [answered]
### What product are you having troubles with?

Marketplace

### What Console version are you using?

1.100.3

### Description

To safely utilize the Mongo CRUD service I need to be able to securely set the environment variable or provide a configuration for the MONGODB_URL from Vault.   

Options that seem viable:
- Mount an environment variable from Vault by executing an export command at container startup.  (requires a modification to the CMD to execute the export before start of the CRUD process).
- Set a configuration file and provide this in the Command-Line Arguments.  If this is available, I am not sure what the argument or config file is.

What would be the recommended approach in this setup?  We will need to avoid storing the Mongo url in Github for security reasons.

### Actual Outcome

To safely utilize the Mongo CRUD service I need to be able to securely mount the environment variable or provide a configuration from Vault.  

### Expected Outcome

I would like direction on how to mount this private configuration into the CRUD container.

**Comment by mattia-fumo** [comment]
Hi @swageblock,
have you already tried to [mount the variable from secret](https://docs.mia-platform.eu/docs/12.x/console/project-configuration/manage-environment-variables/manage-environment-variables-with-vault#use-vault-variables-inside-a-project)? 

It is the recommended way to use Vault secrets inside a Console Project.

**Reply by swageblock** [reply]
I have not, I will attempt this and report back here.  Thanks for the prompt reply.

**Reply by swageblock** [reply]
I think I may have omitted an important detail in my explanation.  We are using the Enhanced Project Workflow.  Based on documentation, it appears that mounting the environment variable from secret is interpolating values and not mounting them at launch time of the container; this would mean that the value would be in our configuration repo.  Is my understanding correct?

**Reply by fredmaggiowski** [reply, marked as answer]
Hi @swageblock when you define a microservice [environment variable from secret](https://docs.mia-platform.eu/docs/12.x/development_suite/api-console/api-design/services#environment-variable-configuration) you can specify the kubernetes secret that holds the key/value pair to be used.

This way the secret material won't be saved in the configuration repository; you must configure it by using another mean (e.g. using the [Project variables section](https://docs.mia-platform.eu/docs/12.x/console/project-configuration/manage-environment-variables/))

In console the UI should display something like the screenshot attached:

<img width="910" alt="Screenshot 2024-06-10 alle 18 41 37" src="https://github.com/mia-platform/community/assets/7142570/0dd07ca3-f8ac-4e90-9a18-9151e9dce48f">

<img width="675" alt="Screenshot 2024-06-10 alle 18 40 57" src="https://github.com/mia-platform/community/assets/7142570/af2d36bb-a15e-4833-a03e-f316b06b2ac4">
